docs/source/checks/conditional_probabilities.rst

```diff
@@ -147,33 +161,45 @@ Algorithm
 Rule Computation
 ################
 
-In the first phase,
-
-1. Both data sets are transformed: each component of every tuple is replaced
+1. Both data sets are pre-processed: numerical columns are binned and textual columns are
+   embedded.
+2. Both data sets are transformed: each component of every tuple is replaced
    by an attribute-name, attribute-value pair. However, this transformation is
    applied on the fly; we never actually copy the data.
-2. The FP-growth algorithm is used to generate *association rules* for both
+3. The FP-growth algorithm is used to generate *association rules* for both
    data sets. The parameters ``min_support`` and
    ``min_confidence`` are used as described in [Han2000]_ and
    [Agrawal1994]_. The only difference is that both parameters are relative and
    expect ``floats`` between 0 and 1, whereas [Han2000]_ and [Agrawal1994]_
    use an absolute value for ``min_support``.
-3. Association rules exceeding ``min_support`` and ``min_confidence`` in both
+4. Association rules exceeding ``min_support`` and ``min_confidence`` in both
    data sets can be compared directly. For each of those rule-pairs generate an
-   intermediate result rule similar to the form showed above.
-4. If a rule exceeds ``min_support`` and ``min_confidence`` in
+   intermediate result rule similar to the form of the red rules showed above.
+5. If a rule exceeds ``min_support`` and ``min_confidence`` in
    one data set but not in the other, we don't know if this rule does not appear in
    the other data set at all or just does not exceed ``min_support`` and/or
    ``min_confidence``. We therefore scan both data sets one
    more time and count their appearances. This information at hand, we can
    generate the remaining intermediate result rules.
-5. The intermediate result rules are filtered for those exceeding
-   ``min_delta_supports`` and ``min_delta_confidences``.
-
-Rule Compression
-################
 
-The second phase
+Rule Reduction
+##############
+
+6. Intermediate result rules are partitioned in those rules that only appear in
+   the first data set, those rules that only appear in the second data set, and
+   those rules that appear in both data sets and consist of a left and a
+   right side (called mutual rules).
+7. Mutual rules are filtered for those exceeding ``min_delta_supports`` and
+   ``min_delta_confidences`` and sorted in descending order according to the
+   absolute difference of their confidence values and the maximum of their
+   supports of left side values. Rules whose support value is 0 in one data
```

Comment:
maybe briefly explain why they come last

Reply:
I will update the documentation in another pr late. But will keep this in mind